shared_model/interfaces/queries/ordering.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_MODEL_ORDERING_HPP
+#define IROHA_SHARED_MODEL_ORDERING_HPP
+
+#include <boost/optional.hpp>
+
+#include "interfaces/base/model_primitive.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    /**
+     * Class Ordering provides description of data ordering in queries.
+     * General note: this is interface.
+     */
+    class Ordering : public ModelPrimitive<Ordering> {
+    public:
+      /**
+       * Field name for ordering entry.
+       */
+      enum struct Field : size_t{
+        kCreatedTime = 0,
+        kPosition,
+        //--------------
+        kMaxValueCount,
+      };
+
+      /**
+       * Ordering direction for each field.
+       */
+      enum struct Direction : size_t{
+        kAscending = 0,
+        kDescending,
+        //--------------
+        kMaxValueCount,
+      };
+
+      /**
+       * Ordering entry - the description of the ordering for the field.
+       */
+      struct OrderingEntry {
+        Field field;
+        Direction direction;
+      };
+
+      /**
+       * Insert - stores field and direction entry uniquely. The
+       * insertion order determines the ordering priority.
+       * @return the insertion result(true - inserted, false - skipped).
+       */
+      virtual bool insert(Field field, Direction direction) = 0;
+
+      /**
+       * Reset - drops all saved data.
+       */
+      virtual void reset() = 0;
+
+      /**
+       * Get - returns data sorted in the insertion order.
+       */
+      virtual void get(OrderingEntry const*& orderingEntry, size_t& count) const = 0;
```

Comment:
here and in other places, please provide docstring for params, including in/out, because it is not trivial from the first look